dashboard/app/controllers/reference_guides_controller.rb

```diff
@@ -15,6 +15,23 @@ def show
     render :not_found unless params[:course_course_name] && params[:key]
   end
 
+  # PATCH /courses/:course_name/guides/:key
+  def update
+    @reference_guide.update!(reference_guide_params.except(:key, :course_course_name))
+    @reference_guide.write_serialization
+    render json: @reference_guide.summarize_for_edit.to_json
+  end
+
+  # DELETE /courses/:course_name/guides/:key
+  def destroy
+    @reference_guide.destroy
+  end
+
+  # GET /courses/:course_name/guides/:key/edit
+  def edit
+    render :not_found unless params[:course_course_name] && params[:key]
```

Comment:
any idea why `course` is repeated in `params[:course_course_name]`, but `key` is not referenced as `params[:course_key]`? 

if it could get rid of the duplication, I would suggest changing the route to `/courses/:name/`.

Reply:
i believe its due to the way i'm nesting ref guides inside courses 
![image](https://user-images.githubusercontent.com/82416901/158891923-7b20cf38-d7dd-4545-be6e-4b3a064b30d5.png)
I could split them out and have them own their own course_name param (i think? it doesn't seem like `resources` is able to handle that)

Reply:
oh I see... the `course_name` param is named within `resources :courses`. I think this is probably not worth cracking open then -- I'm fine with how you have it.